lib/msf/core/data_store.rb

```diff
@@ -216,6 +219,22 @@ def from_file(path, name = 'global')
     end
   end
 
+  def merge!(other)
+    @options.merge!(other.options)
```

Comment:
Since you are adding `attr_accessor`s for these, you should probably reference them as `self.options...` etc.